apps/prairielearn/src/lib/editors.ts

```diff
@@ -263,18 +310,13 @@ export abstract class Editor {
               env: gitEnv,
             });
             job.data.saveSucceeded = true;
+
+            // Clean up to remove any empty directories that might have been
+            // left behind by operations like renames.
+            await cleanAndResetRepository(this.course, `origin/${this.course.branch}`, gitEnv, job);
```

Comment:
There's a change in behavior here. Previously, if the push failed, we'd still clean-and-reset before syncing. Now, if pushing fails, we'll end up syncing what was committed to disk, even though it wasn't successfully pushed.

I believe you want to wrap the `git push` in a `try` and execute `cleanAndResetRepository` in a corresponding `finally`.